src/core/Menu/MenuItem.tsx

```diff
@@ -40,10 +44,18 @@ export type MenuItemProps = {
    * @default 'menuitem'
    */
   role?: string;
+  /**
+   * Items to be shown in the submenu when hovered over the item.
+   */
+  subMenuItems?: JSX.Element[];
   /**
    * Content of the menu item.
    */
   children?: React.ReactNode;
+  /**
+   * Parent `MenuItem` reference. Used in sub-menus.
+   */
+  parentMenuItemRef?: React.RefObject<HTMLLIElement>;
```

Comment:
I don't like this prop myself as it is exposed to the users. I have seen other libs are using `Context` for these kind of stuff so it is not visible for the user.

Reply:
After seeing Context in action in Slider PR (there it was not really needed), I am thinking if we could check more and maybe apply here? Exactly for the same reason to not to show internal stuff to the user...
@mayank99  thoughts?

Reply:
I was actually looking at context myself yesterday. Should have something working by the end of the day

Reply:
I've been playing around for a few hours and can't seem to do it properly yet.

I think I will need to move the submenu logic back into `Menu` and share the state there in the context.

Reply:
I ran context implementation and it seemed to work same as before.. There is something I havent noticed?
But thinking from logical perspective, having a context of the menu might help us solve issues when there is  multiple sub menus opened...

Reply:
What I have pushed works the same as before, so it also has the same problems: opening multiple submenus, and the hack for closing.

I was trying to improve it by moving the logic and context from MenuItem to Menu, but it got really messy and was not working properly. ☹️

Reply:
hmmm..
I think this looks better than the prop.
I would have guessed keeping context on whole menu should work better, but if you say its not.. idk. we could either spend some time on it together to see, or we could go forward with this approach and check up later (should not be breaking change as its inner handling)

Reply:
> What I have pushed works the same as before, so it also has the same problems: opening multiple submenus, and the hack for closing.

I was able to fix both of these by adding event listeners to the a new wrapper div around sub menu.